prost-types/src/datetime.rs

```diff
@@ -330,6 +330,9 @@ fn parse_two_digit_numeric(s: &str) -> Option<(u8, &str)> {
     if s.len() < 2 {
         return None;
     }
+    if s.starts_with('+') {
```

Comment:
The `digits.parse` function will eventually call [`u8::from_str_radix`]([from_str](https://doc.rust-lang.org/std/primitive.u8.html#method.from_str_radix)), which allows to an optional `+` and then only digits. So checking for the `+` as the first character is enough to get all the weird cases of `parse`.

Reply:
yep, that seems to be the case!